coreax/kernel.py

```diff
@@ -483,6 +474,15 @@ def calculate_kernel_matrix_row_sum(
         num_data_points = len(x)
         kernel_row_sum = jnp.zeros(num_data_points)
 
+        # Validate sensible inputs have been given
+        max_size = max(0, max_size)
+        try:
+            range(0, num_data_points, max_size)
```

Comment:
You create a range here to check it is valid and then create a new one on line 487, why not use this one there?

Reply:
Agreed and updated to re-use this one